From 1f186f42b77e4cf9e04ff75c7f5748fce062ea0a Mon Sep 17 00:00:00 2001
From: Ayush Ojha <ayushojzha@gmail.com>
Date: Fri, 27 Mar 2026 12:02:26 +0000
Subject: [PATCH] Use C++17 and the conforming MSVC preprocessor

Backport the build-setting changes from
https://github.com/NVlabs/tiny-cuda-nn/pull/535. The matching CUTLASS
revision is supplied separately by the feedstock recipe.
---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bc66e13..a808c70 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,9 +73,10 @@ endif()
 if (MSVC)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /D_CRT_SECURE_NO_WARNINGS")
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:preprocessor")
 endif()
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
@@ -134,7 +135,7 @@ function(TCNN_AUTODETECT_CUDA_ARCHITECTURES OUT_VARIABLE)
 	endif()
 endfunction()
 
-set(CMAKE_CUDA_STANDARD 14)
+set(CMAKE_CUDA_STANDARD 17)
 set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 set(CMAKE_CUDA_EXTENSIONS OFF)
 set(CUDA_LINK_LIBRARIES_KEYWORD PUBLIC)
@@ -278,6 +279,7 @@ endif()
 
 if (MSVC)
 	list(APPEND CUDA_NVCC_FLAGS "-Xcompiler=/bigobj")
+	list(APPEND CUDA_NVCC_FLAGS "-Xcompiler=/Zc:preprocessor")
 else()
 	list(APPEND CUDA_NVCC_FLAGS "-Xcompiler=-Wno-float-conversion")
 	list(APPEND CUDA_NVCC_FLAGS "-Xcompiler=-fno-strict-aliasing")
-- 
2.51.0
